File: fix_home.py
import re

with open('pages/Home.tsx', 'r') as f:
    code = f.read()

if 'import { Link }' not in code:
    code = code.replace("import React, { useState } from 'react';", "import React, { useState } from 'react';\nimport { Link } from 'react-router-dom';")

# Remove handleScroll function
code = re.sub(r'  const handleScroll = \(.*?\).*?\}\;\n\n', '', code, flags=re.DOTALL)

# Replace <a href="#booking" onClick={(e) => handleScroll(e, 'booking')} className="...">...</a>
# with <Link to="/booking" className="...">...</Link>
def repl(m):
    return f'<Link to="/{m.group(1)}"{m.group(2)}>{m.group(3)}</Link>'

code = re.sub(r'<a\s+href="#(.*?)"\s*onClick=\{.*?handleScroll.*?\}\s*([\s\S]*?)>([\s\S]*?)</a>', repl, code)

with open('pages/Home.tsx', 'w') as f:
    f.write(code)

File: fix_home.js
const fs = require('fs');
let code = fs.readFileSync('pages/Home.tsx', 'utf8');

// Import Link
if (!code.includes('import { Link }')) {
  code = code.replace(/import React, \{ useState \} from 'react';/, "import React, { useState } from 'react';\nimport { Link } from 'react-router-dom';");
}

// Remove handleScroll function
code = code.replace(/const handleScroll = \([\s\S]*?\}\;\n/m, '');

// Replace all links to hash with react-router Links
code = code.replace(/<a\s+href="#(.*?)"[\s\S]*?onClick=\{\(e\) => handleScroll\(e, '.*?'\)\}([\s\S]*?)>([\s\S]*?)<\/a>/gm, '<Link to="/$1"$2>$3</Link>');

// We also have some missing onClick? Let's check if there are other href="#..." without handleScroll
// The script above assumes all have onClick={(e) => handleScroll(e, 'something')}
fs.writeFileSync('pages/Home.tsx', code);
